RELEASING.md

```diff
@@ -0,0 +1,189 @@
+# Releasing
+
+The following steps outline the release process for both new minor versions (e.g.
+releasing the `master` branch as X.Y.0) and patch versions (e.g. releasing the
+`vX.Y` branch as X.Y.Z).
+
+The command examples below assume that the canonical "mongodb" repository has
+the remote name "mongodb". You may need to adjust these commands if you've given
+the remote another name (e.g. "upstream"). The "origin" remote name was not used
+as it likely refers to your personal fork.
+
+It helps to keep your own fork in sync with the "mongodb" repository (i.e. any
+branches and tags on the main repository should also exist in your fork). This
+is left as an exercise to the reader.
+
+## Ensure PHP version compatibility
+
+Ensure that the test suite completes on supported versions of PHP.
+
+## Transition JIRA issues and version
+
+All issues associated with the release version should be in the "Closed" state
+and have a resolution of "Fixed". Issues with other resolutions (e.g.
+"Duplicate", "Works as Designed") should be removed from the release version so
+that they do not appear in the release notes.
+
+Check the corresponding ".x" fix version to see if it contains any issues that
+are resolved as "Fixed" and should be included in this release version.
+
+Update the version's release date and status from the
+[Manage Versions](https://jira.mongodb.org/plugins/servlet/project-config/PHPORM/versions)
+page.
+
+## Update version info
+
+This uses [semantic versioning](https://semver.org/). Do not break
+backwards compatibility in a non-major release or your users will kill you.
+
+Before proceeding, ensure that the `master` branch is up-to-date with all code
+changes in this maintenance branch. This is important because we will later
+merge the ensuing release commits up to master with `--strategy=ours`, which
+will ignore changes from the merged commits.
+
+## Update composer.json
+
+This is especially important before releasing a new minor version.
+
+Ensure that the extension and PHP library requirements, as well as the branch
+alias in `composer.json` are correct for the version being released. For
+example, the branch alias for the 4.1.0 release in the `master` branch should
+be `4.1.x-dev`.
+
+Commit and push any changes:
+
+```console
+$ git commit -m "Update composer.json X.Y.Z" composer.json
+$ git push mongodb
+```
+
+## Tag the release
+
+Create a tag for the release and push:
+
+```console
+$ git tag -a -m "Release X.Y.Z" X.Y.Z
```

Comment:
Keeping `-a` for the time being, but we can consider changing this to `-s` to enforce signed tags.

Reply:
I'm OK with signing the tag.

Reply:
We already have a DRIVERS epic and tracking tickets for this. See [PHPC-1584](https://jira.mongodb.org/browse/PHPC-1584). I'd propose creating a PHPORM ticket and linking that up.